### PR TITLE
NeRFtoGSandBack

### PR DESCRIPTION
**Pull request by Siming-He**
It contains the updates to include NeRFtoGSandBack ([https://arxiv.org/abs/2405.09717](https://arxiv.org/abs/2405.09717)) in nerfstudio. Thanks for your time in advance!

### CODE REVIEW

**Comment by maturk** [comment]
Can you update your branch with latest main?

File: docs/nerfology/methods/nerf2gs2nerf.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# From NeRFs to Gaussian Splats, and Back
+
+This is the implementation of [From NeRFs to Gaussian Splats, and Back](https://arxiv.org/abs/2405.09717); An efficient procedure to convert back and forth between NeRF and GS, and thereby get the best of both approaches. New dataset in the paper can be downloaded from [this google drive link](https://drive.google.com/drive/folders/1xvbONL4EVgHxaHMsV101455l_jNgyaUM?usp=sharing).
+
+<video id="teaser" muted autoplay playsinline loop controls width="100%">
```
Comment (maturk) [inline docs/nerfology/methods/nerf2gs2nerf.md:5]: @Siming-He, really interesting project. I don't fully understand the demo video though, can you briefly describe what the left and right columns depict, mainly the right column. It switches from nerf-sh to nerf-gs (what exactly does that mean)? SH is the nerf variant and then you convert it to the Nerf-GS variant or do you train again to generate the Nerf-GS variant?

**Comment by ichsan2895** [comment]
Interesting project
But how to adjust conversion time from Nerf-SH to NerfGS? It seems the longer time for conversion, the result is way better

**Comment by ichsan2895** [comment]
How to fix it?
```
>> ns-export-nerfsh --load-config ./outputs/unnamed/nerfsh/2024-05-19_104655/config.yml \
    --output-dir ./renders/ --num-points 2000000 \
    --remove-outliers True --normal-method open3d --use_bounding_box False

Variable resolution, using variable_res_collate
Started threads
Setting up evaluation dataset...
Caching all 2 images.
Loading data batch ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Loading data batch ━╸━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   4% -:--:--Loading latest checkpoint from load_dir
Loading data batch ━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━  35% 0:00:03✅ Done loading checkpoint from outputs/unnamed/nerfsh/2024-05-19_104655/nerfstudio_models/step-000012000.ckpt
Loading data batch ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:03-6m0:00:03
☁ Computing Point Cloud ☁ ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 00:4200:0100:02
Cleaning Point Cloud

✅ Cleaning Point Cloud
Estimating Point Cloud Normals

✅ Estimating Point Cloud Normals
✅ Generated 1999372
RPly: Failed writing scale_2 of vertex 7390 (binary output: float)
RPly: Failed writing scale_2 of vertex 7390 (binary output: float)
RPly: Failed writing nz of vertex 41646 (binary output: float)
RPly: Failed writing nz of vertex 41646 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 41646 (binary output: float)
RPly: Failed writing z of vertex 47628 (binary output: float)
RPly: Failed writing z of vertex 47628 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 47628 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 53230 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 53230 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 53231 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 145122 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 145122 (binary output: float)
RPly: Failed writing opacity of vertex 145123 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 150968 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 150968 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 174010 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 174010 (binary output: float)
RPly: Failed writing ny of vertex 174011 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 185247 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 185247 (binary output: float)
RPly: Failed writing y of vertex 185248 (binary output: float)
RPly: Failed writing f_rest_11 of vertex 185337 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 185337 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 185396 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 185396 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 185396 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 214038 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 214038 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 214038 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 219856 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 219856 (binary output: float)
RPly: Failed writing f_rest_37 of vertex 219856 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 242796 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 242796 (binary output: float)
RPly: Failed writing f_dc_1 of vertex 242796 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 254257 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 254257 (binary output: float)
RPly: Failed writing f_rest_14 of vertex 254257 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 260115 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 260115 (binary output: float)
RPly: Failed writing f_rest_17 of vertex 260115 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 266057 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 266057 (binary output: float)
RPly: Failed writing f_rest_2 of vertex 266057 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 288974 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 288974 (binary output: float)
RPly: Failed writing f_rest_13 of vertex 288974 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 300444 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 300444 (binary output: float)
RPly: Failed writing f_rest_7 of vertex 300444 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 311629 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 311629 (binary output: float)
RPly: Failed writing f_rest_21 of vertex 311629 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 317558 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 317558 (binary output: float)
RPly: Failed writing f_rest_25 of vertex 317558 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 340533 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 340533 (binary output: float)
RPly: Failed writing f_rest_40 of vertex 340533 (binary output: float)
RPly: Failed writing opacity of vertex 357454 (binary output: float)
RPly: Failed writing opacity of vertex 357454 (binary output: float)
RPly: Failed writing scale_1 of vertex 357454 (binary output: float)
RPly: Failed writing scale_2 of vertex 357543 (binary output: float)
RPly: Failed writing nz of vertex 374934 (binary output: float)
RPly: Failed writing nz of vertex 374934 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 374934 (binary output: float)
RPly: Failed writing z of vertex 392313 (binary output: float)
RPly: Failed writing z of vertex 392313 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 392313 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 403837 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 403837 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 403838 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 415207 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 415207 (binary output: float)
RPly: Failed writing opacity of vertex 415208 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 432387 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 432387 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 449516 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 449516 (binary output: float)
RPly: Failed writing ny of vertex 449517 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 449555 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 449555 (binary output: float)
RPly: Failed writing y of vertex 449556 (binary output: float)
RPly: Failed writing f_rest_11 of vertex 449775 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 449775 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 513051 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 513051 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 513051 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 541851 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 541851 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 541851 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 587884 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 587884 (binary output: float)
RPly: Failed writing f_rest_37 of vertex 587884 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 616537 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 616537 (binary output: float)
RPly: Failed writing f_dc_1 of vertex 616537 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 622486 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 622486 (binary output: float)
RPly: Failed writing f_rest_14 of vertex 622486 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 628030 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 628030 (binary output: float)
RPly: Failed writing f_rest_17 of vertex 628030 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 639674 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 639674 (binary output: float)
RPly: Failed writing f_rest_2 of vertex 639674 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 662708 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 662708 (binary output: float)
RPly: Failed writing f_rest_13 of vertex 662708 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 668660 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 668660 (binary output: float)
RPly: Failed writing f_rest_7 of vertex 668660 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 680001 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 680001 (binary output: float)
RPly: Failed writing f_rest_21 of vertex 680001 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 697181 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 697181 (binary output: float)
RPly: Failed writing f_rest_25 of vertex 697181 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 743144 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 743144 (binary output: float)
RPly: Failed writing f_rest_40 of vertex 743144 (binary output: float)
RPly: Failed writing opacity of vertex 754633 (binary output: float)
RPly: Failed writing opacity of vertex 754633 (binary output: float)
RPly: Failed writing scale_1 of vertex 754633 (binary output: float)
RPly: Failed writing scale_2 of vertex 777555 (binary output: float)
RPly: Failed writing nz of vertex 806384 (binary output: float)
RPly: Failed writing nz of vertex 806384 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 806384 (binary output: float)
RPly: Failed writing z of vertex 812116 (binary output: float)
RPly: Failed writing z of vertex 812116 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 812116 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 812159 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 812159 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 812160 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 817663 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 817663 (binary output: float)
RPly: Failed writing opacity of vertex 817664 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 817707 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 817707 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 829303 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 829303 (binary output: float)
RPly: Failed writing ny of vertex 829304 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 835272 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 835272 (binary output: float)
RPly: Failed writing y of vertex 835273 (binary output: float)
RPly: Failed writing f_rest_11 of vertex 852438 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 852438 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 863827 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 863827 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 863827 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 863889 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 863889 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 863889 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 864135 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 864135 (binary output: float)
RPly: Failed writing f_rest_37 of vertex 864135 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 886787 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 886787 (binary output: float)
RPly: Failed writing f_dc_1 of vertex 886787 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 898351 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 898351 (binary output: float)
RPly: Failed writing f_rest_14 of vertex 898351 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 898447 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 898447 (binary output: float)
RPly: Failed writing f_rest_17 of vertex 898447 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 915762 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 915762 (binary output: float)
RPly: Failed writing f_rest_2 of vertex 915762 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 915794 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 915794 (binary output: float)
RPly: Failed writing f_rest_13 of vertex 915794 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 950122 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 950122 (binary output: float)
RPly: Failed writing f_rest_7 of vertex 950122 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 961848 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 961848 (binary output: float)
RPly: Failed writing f_rest_21 of vertex 961848 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 996009 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 996009 (binary output: float)
RPly: Failed writing f_rest_25 of vertex 996009 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1007694 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1007694 (binary output: float)
RPly: Failed writing f_rest_40 of vertex 1007694 (binary output: float)
RPly: Failed writing opacity of vertex 1013380 (binary output: float)
RPly: Failed writing opacity of vertex 1013380 (binary output: float)
RPly: Failed writing scale_1 of vertex 1013380 (binary output: float)
RPly: Failed writing scale_2 of vertex 1024617 (binary output: float)
RPly: Failed writing nz of vertex 1041616 (binary output: float)
RPly: Failed writing nz of vertex 1041616 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 1041616 (binary output: float)
RPly: Failed writing z of vertex 1047566 (binary output: float)
RPly: Failed writing z of vertex 1047566 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 1047566 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 1047621 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 1047621 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1047622 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 1083655 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 1083655 (binary output: float)
RPly: Failed writing opacity of vertex 1083656 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 1083772 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 1083772 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 1099326 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 1099326 (binary output: float)
RPly: Failed writing ny of vertex 1099327 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 1099459 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 1099459 (binary output: float)
RPly: Failed writing y of vertex 1099460 (binary output: float)
RPly: Failed writing f_rest_11 of vertex 1116598 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 1116598 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 1122503 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 1122503 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 1122503 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 1145495 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 1145495 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 1145495 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 1151156 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 1151156 (binary output: float)
RPly: Failed writing f_rest_37 of vertex 1151156 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 1168428 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 1168428 (binary output: float)
RPly: Failed writing f_dc_1 of vertex 1168428 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 1181387 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 1181387 (binary output: float)
RPly: Failed writing f_rest_14 of vertex 1181387 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 1191158 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 1191158 (binary output: float)
RPly: Failed writing f_rest_17 of vertex 1191158 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 1191365 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 1191365 (binary output: float)
RPly: Failed writing f_rest_2 of vertex 1191365 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 1198620 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 1198620 (binary output: float)
RPly: Failed writing f_rest_13 of vertex 1198620 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 1248752 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 1248752 (binary output: float)
RPly: Failed writing f_rest_7 of vertex 1248752 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 1254392 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 1254392 (binary output: float)
RPly: Failed writing f_rest_21 of vertex 1254392 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 1271557 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 1271557 (binary output: float)
RPly: Failed writing f_rest_25 of vertex 1271557 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1277508 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1277508 (binary output: float)
RPly: Failed writing f_rest_40 of vertex 1277508 (binary output: float)
RPly: Failed writing opacity of vertex 1294738 (binary output: float)
RPly: Failed writing opacity of vertex 1294738 (binary output: float)
RPly: Failed writing scale_1 of vertex 1294738 (binary output: float)
RPly: Failed writing scale_2 of vertex 1317895 (binary output: float)
RPly: Failed writing nz of vertex 1352299 (binary output: float)
RPly: Failed writing nz of vertex 1352299 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 1352299 (binary output: float)
RPly: Failed writing z of vertex 1369529 (binary output: float)
RPly: Failed writing z of vertex 1369529 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 1369529 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 1375191 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 1375191 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1375192 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 1403894 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 1403894 (binary output: float)
RPly: Failed writing opacity of vertex 1403895 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 1403905 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 1403905 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 1404123 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 1404123 (binary output: float)
RPly: Failed writing ny of vertex 1404124 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 1415599 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 1415599 (binary output: float)
RPly: Failed writing y of vertex 1415600 (binary output: float)
RPly: Failed writing f_rest_11 of vertex 1415633 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 1415633 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 1438371 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 1438371 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 1438371 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 1438376 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 1438376 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 1438376 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 1438440 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 1438440 (binary output: float)
RPly: Failed writing f_rest_37 of vertex 1438440 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 1461528 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 1461528 (binary output: float)
RPly: Failed writing f_dc_1 of vertex 1461528 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 1461762 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 1461762 (binary output: float)
RPly: Failed writing f_rest_14 of vertex 1461762 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 1472976 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 1472976 (binary output: float)
RPly: Failed writing f_rest_17 of vertex 1472976 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 1478692 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 1478692 (binary output: float)
RPly: Failed writing f_rest_2 of vertex 1478692 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 1484476 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 1484476 (binary output: float)
RPly: Failed writing f_rest_13 of vertex 1484476 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 1490051 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 1490051 (binary output: float)
RPly: Failed writing f_rest_7 of vertex 1490051 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 1507193 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 1507193 (binary output: float)
RPly: Failed writing f_rest_21 of vertex 1507193 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 1507421 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 1507421 (binary output: float)
RPly: Failed writing f_rest_25 of vertex 1507421 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1518686 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1518686 (binary output: float)
RPly: Failed writing f_rest_40 of vertex 1518686 (binary output: float)
RPly: Failed writing opacity of vertex 1524332 (binary output: float)
RPly: Failed writing opacity of vertex 1524332 (binary output: float)
RPly: Failed writing scale_1 of vertex 1524332 (binary output: float)
RPly: Failed writing scale_2 of vertex 1535779 (binary output: float)
RPly: Failed writing nz of vertex 1558880 (binary output: float)
RPly: Failed writing nz of vertex 1558880 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 1558880 (binary output: float)
RPly: Failed writing z of vertex 1656403 (binary output: float)
RPly: Failed writing z of vertex 1656403 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 1656403 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 1656411 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 1656411 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1656412 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 1668135 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 1668135 (binary output: float)
RPly: Failed writing opacity of vertex 1668136 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 1673811 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 1673811 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 1673867 (binary output: float)
RPly: Failed writing f_rest_6 of vertex 1673867 (binary output: float)
RPly: Failed writing ny of vertex 1673868 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 1679335 (binary output: float)
RPly: Failed writing f_rest_32 of vertex 1679335 (binary output: float)
RPly: Failed writing y of vertex 1679336 (binary output: float)
RPly: Failed writing f_rest_11 of vertex 1713724 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 1713724 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 1731051 (binary output: float)
RPly: Failed writing f_rest_12 of vertex 1731051 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 1731051 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 1771319 (binary output: float)
RPly: Failed writing f_dc_0 of vertex 1771319 (binary output: float)
RPly: Failed writing f_rest_3 of vertex 1771319 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 1776961 (binary output: float)
RPly: Failed writing f_dc_2 of vertex 1776961 (binary output: float)
RPly: Failed writing f_rest_37 of vertex 1776961 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 1817097 (binary output: float)
RPly: Failed writing f_rest_24 of vertex 1817097 (binary output: float)
RPly: Failed writing f_dc_1 of vertex 1817097 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 1845850 (binary output: float)
RPly: Failed writing f_rest_16 of vertex 1845850 (binary output: float)
RPly: Failed writing f_rest_14 of vertex 1845850 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 1845995 (binary output: float)
RPly: Failed writing f_rest_22 of vertex 1845995 (binary output: float)
RPly: Failed writing f_rest_17 of vertex 1845995 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 1857367 (binary output: float)
RPly: Failed writing f_rest_41 of vertex 1857367 (binary output: float)
RPly: Failed writing f_rest_2 of vertex 1857367 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 1863101 (binary output: float)
RPly: Failed writing f_rest_27 of vertex 1863101 (binary output: float)
RPly: Failed writing f_rest_13 of vertex 1863101 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 1863286 (binary output: float)
RPly: Failed writing f_rest_28 of vertex 1863286 (binary output: float)
RPly: Failed writing f_rest_7 of vertex 1863286 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 1892021 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 1892021 (binary output: float)
RPly: Failed writing f_rest_21 of vertex 1892021 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 1909061 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 1909061 (binary output: float)
RPly: Failed writing f_rest_25 of vertex 1909061 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1909077 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1909077 (binary output: float)
RPly: Failed writing f_rest_40 of vertex 1909077 (binary output: float)
RPly: Failed writing opacity of vertex 1909285 (binary output: float)
RPly: Failed writing opacity of vertex 1909285 (binary output: float)
RPly: Failed writing scale_1 of vertex 1909285 (binary output: float)
RPly: Failed writing scale_2 of vertex 1920590 (binary output: float)
RPly: Failed writing nz of vertex 1920735 (binary output: float)
RPly: Failed writing nz of vertex 1920735 (binary output: float)
RPly: Failed writing f_rest_30 of vertex 1920735 (binary output: float)
RPly: Failed writing z of vertex 1938102 (binary output: float)
RPly: Failed writing z of vertex 1938102 (binary output: float)
RPly: Failed writing f_rest_33 of vertex 1938102 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 1949526 (binary output: float)
RPly: Failed writing f_rest_15 of vertex 1949526 (binary output: float)
RPly: Failed writing f_rest_39 of vertex 1949527 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 1972204 (binary output: float)
RPly: Failed writing f_rest_4 of vertex 1972204 (binary output: float)
RPly: Failed writing opacity of vertex 1972205 (binary output: float)
✅ Saving Spherical Harmonics
```

File: docs/nerfology/methods/nerf2gs2nerf.md (hunk 1 above)
Comment (Siming-He) [reply docs/nerfology/methods/nerf2gs2nerf.md:5]: Hi, I updated my branch with the latest main. For the video, NeRF-SH is the nerf variant that predicts spherical harmonic coefficients. I directly convert it to NeRF-GS with conversion time 7s (which is Gaussian splats extracted from NeRF-GS). Then I finetune NeRF-GS on the training data (for 3s, 33s, and 153s) to get better NeRF-GS. Thank you for the question! I'll add the explanations into the video.

**Comment by Siming-He** [comment]
> Interesting project But how to adjust conversion time from Nerf-SH to NerfGS? It seems the longer time for conversion, the result is way better

Thank you! In the video, 7s is the time to export NeRFGS directly from NeRF-SH. Then, we finetune the NeRFGS for 3s, 33s and 153s on training data for better results.

**Comment by ichsan2895** [comment]
Any reviewers yet? I am not the developers of Nerfstudio

**Comment by maturk** [comment]
@Siming-He, just one question. Does it make sense to report the "training for 3s, 33s, or 153s" since doesn't the finetuning and amount of seconds highly depend on your computer specs? I wonder if its more meaningful to report how many iterations you retrain or how many images are seen during re-training (since for example GS is trained per-image at a training iteration, so num iterations is equivalent to num of images seen).

**Comment by maturk** [comment]
Great, thanks @Siming-He for answering my questions. LGTM, if you update your branch I can merge it!

**Review by maturk — APPROVED** [review]

**Pull request merged**